src/Arduino_ConnectionHandler.h

```diff
@@ -57,27 +54,46 @@ class ConnectionHandler {
     virtual void init() = 0;
     virtual void check() = 0;
     virtual void update() = 0;
-    virtual unsigned long getTime() = 0;
-    virtual Client &getClient();
-    virtual UDP &getUDP();
 
     virtual NetworkConnectionState getStatus() {
       return netConnectionState;
     }
     virtual void connect();
     virtual void disconnect();
-    virtual void addCallback(NetworkConnectionEvent const event, OnNetworkEventCallback callback);
-    virtual void addConnectCallback(OnNetworkEventCallback callback);
-    virtual void addDisconnectCallback(OnNetworkEventCallback callback);
-    virtual void addErrorCallback(OnNetworkEventCallback callback);
 
-  private:
-    OnNetworkEventCallback  _on_connect_event_callback,
-                            _on_disconnect_event_callback,
-                            _on_error_event_callback;
+    void addCallback(NetworkConnectionEvent const event, OnNetworkEventCallback callback) {
```

Comment:
Please no inlined functions - they make the header file very hard to read - please extract them into a `Arduino_ConnectionHandler.cpp` file.

Reply:
Ok, I created the Arduino_ConnectionHandler.cpp file and removed the implementation from the header.